core/src/main/scala/org/apache/spark/scheduler/EventLoggingListener.scala

```diff
@@ -230,6 +230,13 @@ private[spark] class EventLoggingListener(
       }
     }
     fileSystem.rename(new Path(logPath + IN_PROGRESS), target)
+    // touch file to ensure modtime is current across those filesystems where rename()
+    // does not set it, -and which support setTimes(); it's a no-op on most object stores
+    try {
+      fileSystem.setTimes(target, System.currentTimeMillis(), -1)
+    } catch {
+      case e: Exception => logDebug(s"failed to set time of $target", e)
+    }
```

Comment:
It doesn't impact the new incomplete caching work but it does ensure that all filesystems have the same outcome on the `stop()` call: the final file has the modtime of when the log was closed. Without that,  Posix and HDFS would be different. Because there's other timestamp checking going on in the original scan, it may be safest to leave it in.